services/database/db.js

```diff
@@ -0,0 +1,17 @@
+import mongoose from "mongoose";
+mongoose.Promise = global.Promise;
+
+let isConnected;
+const url = `mongodb://${process.env.MONGO_USER}:${process.env.MONGO_PASSWORD}@ds157136.mlab.com:57136/hackone`;
+
+export const connectToDatabase = () => {
+  if (isConnected) {
```

Comment:
lets not use a stored value that is not kept in sync:

https://mongoosejs.com/docs/api/mongoose.html#mongoose_Mongoose-connection

```
if (mongoose.connection.readyState === 1)
```

Although I'm wondering how much we need to do this... can we call it once during the execution of our function (outside the function, not inside) (`keepAlive` is on by default: https://mongoosejs.com/docs/connections.html#keepAlive)

The commands have buffering enabled by default (https://mongoosejs.com/docs/connections.html#buffering) so we dont need to wait for it to connect before moving on here.

Reply:
Not able to find an example online that calls the connection outside of the lambda function, the official recommendation from Mongo seems to also advocate using connection inside the function. At the moment it seems okay.

https://docs.atlas.mongodb.com/best-practices-connecting-to-aws-lambda/

Reply:
interesting find :)